packages/lesswrong/server/userJobAdCron.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import moment from 'moment';
+import { JOB_AD_DATA } from '../components/ea-forum/TargetedJobAd';
+import { addCronJob } from './cronUtil';
+import UserJobAds from '../lib/collections/userJobAds/collection';
+import { Users } from '../lib/collections/users/collection';
+import uniq from 'lodash/fp/uniq';
+import { wrapAndSendEmail } from './emails/renderEmail';
+import './emailComponents/EmailJobAdReminder';
+import { Components } from './vulcan-lib';
+import { loggerConstructor } from '../lib/utils/logging';
+
+
+addCronJob({
+  name: 'sendJobAdReminderEmails',
+  interval: `every 1 day`,
+  async job() {
+    const logger = loggerConstructor(`cron-sendJobAdReminderEmails`)
+
+    // Find all the job ads where the deadline is one week away
+    const jobNames = Object.keys(JOB_AD_DATA).filter(jobName => {
+      const deadline = JOB_AD_DATA[jobName].deadline
+      return deadline && moment().add(1, 'week').isSameOrBefore(deadline) && moment().add(1, 'week').add(1, 'day').isAfter(deadline)
+    })
+    if (!jobNames.length) {
+      logger('No eligible jobs found')
+      return
+    }
+
+    // Find all the user reminders set for these jobs
+    const userJobAds = await UserJobAds.find({
+      jobName: {$in: jobNames},
```

Comment:
There's no index for either of these fields

Reply:
Thanks - added